### PR TITLE
Integrate lore library

### PR DESCRIPTION
**Pull request by kwy95**

### CODE REVIEW

File: etc/liblore-copy.sh [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+# that follows this pattern:
+#
+#  Author, email, version, total patches, patch title, link
+#
+#Note: To separate those elements, we use the variable SEPARATOR_CHAR, which
+#can be a ',' but by default, we use 'Æ'. We used ',' in the example for make
+#it easy to undertand.
+declare -ag list_of_mailinglist_patches
+
+declare -r BASE_PATH_MBOX_DOWNLOAD="$PWD/mbox_download"
```
Comment (rodrigosiqueira) [inline etc/liblore-copy.sh:17]: Use `{PWD}`

File: src/lib/web_access.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# @url        Target url
+# @output     Name of the output file
+# @cache_path Save downloaded page to a specific path
+function download()
+{
+  local url="$1"
+  local output=${2:-'page.xml'}
+  local flag="$3"
+  local output_path
+
```
Comment (rodrigosiqueira) [inline src/lib/web_access.sh:17]: Check if `url` is empty.

File: src/lib/web_access.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# This file handles any web access
+
+include "${KW_LIB_DIR}/kwio.sh"
+include "${KW_LIB_DIR}/kwlib.sh"
+
+# Download a webpage.
+#
+# @url        Target url
+# @output     Name of the output file
+# @cache_path Save downloaded page to a specific path
```
Comment (rodrigosiqueira) [inline src/lib/web_access.sh:10]: @flag?

File: kw [hunk 4]
```diff
@@ -61,6 +61,14 @@ function kw()
   mkdir -p "$KW_CACHE_DIR"
 
   case "$action" in
+    lore)
+      (
+        include "$KW_LIB_DIR/lib/lore.sh"
```
Comment (rodrigosiqueira) [inline kw:66]: `/KW_LIB_DIR/{KW_LIB_DIR}/`

File: src/lib/lore.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
+# This file is the library that handles the "back end" of interacting with the
+# kernel lore archives. it handles fecthing, listing and downloading of patches
+# sent to the public mailing lists.
+
+include "${KW_LIB_DIR}/kwlib.sh"
+include "${KW_LIB_DIR}/lib/web_access.sh"
+
+declare -g LORE_URL='https://lore.kernel.org/'
```
Comment (rodrigosiqueira) [inline src/lib/lore.sh:8]: -r?

File: src/lib/lore.sh [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+declare -Ag mailing_lists
```
Comment (rodrigosiqueira) [inline src/lib/lore.sh:9]: available_lore_lists? lore_lists? lore_mailng_list_names?

File: src/lib/lore.sh [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+function retrieve_available_mailing_lists()
```
Comment (rodrigosiqueira) [inline src/lib/lore.sh:11]: Add documentation

File: src/lib/lore.sh [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+{
+  local main_page='lore_main_page.html'
+  local flag="$1"
+  local index=''
+  local pre_processed
+
+  flag=${flag:-'SILENT'}
+
+  download "$LORE_URL" "$main_page" "$flag"
```
Comment (rodrigosiqueira) [inline src/lib/lore.sh:20]: We need to check if the `download` operation failed?

File: tests/lib/lore_test.sh [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+function test_retrieve_available_mailing_lists()
+{
+  local index
+  local -A expected=(
+    ['all']='All of lore.kernel.org'
+    ['lkml']='LKML Archive on lore.kernel.org'
+    ['linux-parisc']='Linux-parisc Archive on lore.kernel.org'
+    ['ddprobe']='List used for roundtrip monitoring'
+    ['linux-pm']='Linux-PM Archive on lore.kernel.org'
```
Comment (rodrigosiqueira) [inline tests/lib/lore_test.sh:23]: amd-gfx and dri-devel.

File: etc/liblore-copy.sh (hunk 1 above)
Comment (kwy95) [reply etc/liblore-copy.sh:17]: This file is only here as temporary reference, so it will be deleted later.

**Comment by rodrigosiqueira** [comment]
I'm going to close this PR for now.

**Pull request closed without merging**